project-folder/model.py

```diff
@@ -1,53 +1,86 @@
 import numpy as np
-
+from keras.callbacks import ReduceLROnPlateau
 from keras.models import Sequential
 from keras.layers import Dense, Dropout, Flatten
 from keras.layers import Conv2D, MaxPooling2D
+from keras.preprocessing.image import ImageDataGenerator
+from sklearn.model_selection import train_test_split
 
 
 class Model:
     def __init__(self):
         # Model parameters
-        self.epochs = 10
-        self.batch_size = 32
-        self.num_class = 6
+        self.epochs = 50
+        self.batch_size = 100
+        self.num_class = 7
         self.num_rows = 28
         self.num_cols = 28
+        self.learning_rate_reduction = ReduceLROnPlateau(
+                                        monitor='val_accuracy', patience=3,
+                                        verbose=1, factor=0.5,
+                                        min_lr=0.00001)
+
+        self.datagen = ImageDataGenerator(
+                        featurewise_center=False,
+                        samplewise_center=False,
+                        featurewise_std_normalization=False,
+                        samplewise_std_normalization=False,
+                        zca_whitening=False, rotation_range=10,
+                        zoom_range=0.1, width_shift_range=0.1,
+                        height_shift_range=0.1, horizontal_flip=False,
+                        vertical_flip=False)
 
         # Data
         self.X: np.array
         self.y: np.array
 
         self.model: Sequential
 
+    def split(self):
+        self.X_train, self.X_test, self.y_train, self.y_test =
+        train_test_split(self.X, self.y, test_size=0.20, random_state=42)
+        self.X_train, self.X_validate, self.y_train, self.y_test
+        = train_test_split(self.X_train, self.y_train, test_size=0.20,
```

Comment:
I think that's too much data given to test/validation, esp since validation is only used to tune the learning rate. Training data is only 64% here. I would suggest no more than 10% for validation.